javascript-data-model/main.js

```diff
@@ -0,0 +1,45 @@
+/* exported orderHistory */
+
+var orderHistory = [
+  {
+    name: "JavaScript for impatient programmers",
+    type: "book",
+    by: "Rauschmayer, Dr. Axel",
+    price: 31.55,
+    deliveredDate: 08082020,
+    returnByDate: 09072020
+  },
+  {
+    name: "The Timeless Ways of Building",
+    type: "book",
+    by: "Alexander, Christopher",
+    price: 41.33,
+    deliveredDate: 07202020,
+    returnByDate: 0819202
+  },
+  {
+    name: "Gamecube Controller Adapter",
+    type: ["wii U", "PC"],
+    by: "Super Smash Bros Switch",
+    price: 15.98,
+    deliveredDate: 07072020,
+    returnByDate: 08052020
+  },
+  {
+    name: "Gamecube Controller",
+    type: "Super Smash Bros. Edition",
+    by: "Nintendo Switch",
+    price: 94.95,
+    deliveredDate: 07052020,
+    returnByDate: 08042020
+  },
+  {
```

Comment:
When this data is outputted, how is it supposed to know that this and the item before it were in the same order?

Reply:
This is still an issue. As of now, you have an array of items that were ordered, but you really need an array of the details for the order. How does the browser know that this and the item above are in the same order?